static/js/admin/src/CreateWorkObjectForm.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Implements functionality and UI for the generic Ingest Package form
+ */
+define('CreateWorkObjectForm', [ 'jquery', 'jquery-ui', 'underscore', 'RemoteStateChangeMonitor', 'tpl!../templates/admin/createWorkObjectForm',
+		'ModalCreate', 'ModalLoadingOverlay', 'ConfirmationDialog', 'AbstractFileUploadForm', 'ResultObject'],
+		function($, ui, _, RemoteStateChangeMonitor, workObjectTemplate, ModalCreate, ModalLoadingOverlay, ConfirmationDialog, AbstractFileUploadForm, ResultObject) {
+	
+	var defaultOptions = {
+		title : 'Add New Work',
+		createFormTemplate : workObjectTemplate
+	};
+
+	function CreateWorkObjectForm(options) {
+		this.options = $.extend({}, AbstractFileUploadForm.prototype.getDefaultOptions(), defaultOptions, options);
+	}
+	
+	CreateWorkObjectForm.prototype.constructor = CreateWorkObjectForm;
+	//CreateWorkObjectForm.prototype = Object.create( AbstractFileUploadForm.prototype );
+	CreateWorkObjectForm.prototype.open = function(resultObject) {
+		var self = this;
+		this.closed = false;
+
+		var dialogBox = new ModalCreate(this.options);
+		var formContents = dialogBox.formContents(resultObject);
+
+		this.dialog = $("<div class='containingDialog'>" + formContents + "</div>");
+		this.$form = this.dialog.first();
+		this.dialog.dialog = dialogBox.modalDialog(this.dialog, self);
+
+		var containingDialog = $(".containingDialog");
+		var iframe = $("#work_submission_form");
+		$("select[name='name']").on("change", function() {
+			var ajaxIcon = $(".loading-icon");
+
+			var formUrl = location.href.split("/");
+			var protocol = formUrl[0];
+			var collectionId = iframe.attr("title");
+			var host, subDirectory;
+
+			if (/cdr/.test(formUrl[2])) {
```

Comment:
Hmm, I'd really rather not environmental details like this in our view code.  I would rather if we added a property to server.properties that specifies the base uri for the forms app in this environment, which would get passed down to this form.  forms.site.baseUri or something along those lines.